### ISSUE TITLE
Memory leak when selecting a large number of rows

### ISSUE THREAD

**Issue by yassenb** [JetBrains/Exposed#1302]
Doing a simple `EntityClass.find` results in a memory leak (or perhaps more correctly bloat) when iterating through millions of records. The culprit is `Transaction.entityCache` which only gets filled and never freed during the iteration. As a result I'm forced to use a workaround like:
```
EntityClass.find { ... }.chunked(100_000).forEach {
    // read the chunk
    entityCache.data.clear()
}
```
instead of the simple
```
EntityClass.find { ... }.forEach {
    // read the entity
}
```
There should be a way to disable the cache or configure its size so that it doesn't grow forever.

**Comment by yassenb** [comment]
Interesting that you would label the ability to work with real life databases with more than a few thousand records in them without crashing due to running out of memory an "enhancement".

**Comment by bdshadow** [comment]
I just guess from the reply of @Tapac [here](https://stackoverflow.com/questions/53260153/caching-with-kotlin-exposed) that it was supposed that cache would be used for "mostly dictionary-like" entities, which contain a rather limited amount of data. That's why it is an enhancement/new feature to make it possible to cache big amounts of data without requiring user to clear cache him/herself

**Comment by yassenb** [comment]
It may be "supposed" that caching would be used in certain scenarios but at the moment it's used everywhere and there's no way to disable it. That's what this issue is about.
As per "requiring user to clear cache him/herself" - the workaround I've found is hardly public API. I found it only after reading the source code. What's more, the workaround splits the sequence/iterable in chunks which doesn't preserve the original behaviour which is a lot more convenient and optimal.
